tests/components/awair/test_config_flow.py

```diff
@@ -0,0 +1,184 @@
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["errors"] == {CONF_ACCESS_TOKEN: "auth"}
+
+
+async def test_unexpected_api_error(hass):
+    """Test that we abort on generic errors."""
+
+    with patch("python_awair.AwairClient.query", side_effect=AwairError()):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "unknown"
+
+
+async def test_duplicate_error(hass):
+    """Test that errors are shown when adding a duplicate config."""
+
+    with patch(
+        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
+    ), patch(
+        "homeassistant.components.awair.sensor.async_setup_entry", return_value=True,
+    ):
+        MockConfigEntry(
+            domain=DOMAIN, unique_id=CONFIG_ENTRY_UNIQUE_ID, data=CONFIG
+        ).add_to_hass(hass)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "already_configured"
+
+
+async def test_no_devices_error(hass):
+    """Test that errors are shown when the API returns no devices."""
+
+    with patch(
+        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, NO_DEVICES_FIXTURE]
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "no_devices"
+
+
+async def test_import(hass):
+    """Test config.yaml import."""
+
+    with patch(
+        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
+    ), patch(
+        "homeassistant.components.awair.sensor.async_setup_entry", return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={CONF_ACCESS_TOKEN: CONFIG[CONF_ACCESS_TOKEN]},
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "foo@bar.com (32406)"
+        assert result["data"][CONF_ACCESS_TOKEN] == CONFIG[CONF_ACCESS_TOKEN]
+        assert result["result"].unique_id == CONFIG_ENTRY_UNIQUE_ID
+
+
+async def test_import_aborts_if_configured(hass):
+    """Test config import doesn't re-import unnecessarily."""
+
+    with patch(
+        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
+    ), patch(
+        "homeassistant.components.awair.sensor.async_setup_entry", return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
```

Comment:
We can create a mock entry for the first existing entry instead.

Reply:
df6b56e8a2